flint/flint-spark-integration/src/main/scala/org/apache/spark/sql/flint/storage/FlintQueryCompiler.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.apache.spark.sql.flint.storage
+
+import org.apache.spark.sql.catalyst.expressions.Literal
+import org.apache.spark.sql.connector.expressions.{Expression, FieldReference, LiteralValue}
+import org.apache.spark.sql.connector.expressions.filter.{And, Predicate}
+import org.apache.spark.sql.types.{StringType, StructField, StructType}
+
+/**
+ * Todo. find the right package.
+ */
+case class FlintQueryCompiler(schema: StructType) {
+
+  /**
+   * Using AND to concat predicates. Todo. If spark spark.sql.ansi.enabled = true, more expression
+   * defined in V2ExpressionBuilder could be pushed down.
+   */
+  def compile(predicates: Array[Predicate]): String = {
+    if (predicates.isEmpty) {
+      return ""
+    }
+    compile(predicates.reduce(new And(_, _)))
+  }
+
+  /**
+   * Compile Expression to Flint query string.
+   *
+   * @param expr
+   *   Expression.
+   * @return
+   *   empty if does not support.
+   */
+  def compile(expr: Expression, quoteString: Boolean = true): String = {
+    expr match {
+      case LiteralValue(value, dataType) =>
+        if (quoteString && dataType == StringType) {
+          s""""${Literal(value, dataType).toString()}""""
+        } else {
+          Literal(value, dataType).toString()
+        }
+      case p: Predicate => visitPredicate(p)
+      case f: FieldReference => f.toString()
+      case _ => ""
+    }
+  }
+
+  /**
+   * Predicate is defined in SPARK filters.scala. Todo.
+   *   1. currently, we map spark contains to OpenSearch match query. Can we leverage more full
+   *      text queries for text field. 2. configuration of expensive query.
+   */
+  def visitPredicate(p: Predicate): String = {
+    val name = p.name()
+    name match {
+      case "IS_NULL" =>
+        s"""{"bool":{"must_not":{"exists":{"field":"${compile(p.children()(0))}"}}}}"""
```

Comment:
np: will be more readable if multi-line string

Reply:
yes, try it before. but it will add space and newline, not easy for test also.